### PR TITLE
simple fly-sync

### PR DESCRIPTION
**Pull request by Chrometurtle**

### CODE REVIEW

File: Game/RainMeadow.GameHooks.cs [hunk 1]
```diff
@@ -29,13 +29,26 @@ private void GameHooks()
...
             
             // Arena specific
             On.GameSession.AddPlayer += GameSession_AddPlayer;
         }
 
+        private void Room_ReadyForAI(On.Room.orig_ReadyForAI orig, Room self)
+        {
+            if (OnlineManager.lobby != null)
+            {
+                if (!RoomSession.map.TryGetValue(self.abstractRoom, out var rs)) return;
```
Comment (henpemaz) [inline Game/RainMeadow.GameHooks.cs:44]: Doesn't run orig if you do it this way. This is a "should never happen" scenario though so probably doesn't matter.

File: Online/State/AbstractCreatureState.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ protected override RealizedPhysicalObjectState GetRealizedState(OnlinePhysicalOb
             if (onlineObject.apo.realizedObject is Player) return new RealizedPlayerState((OnlineCreature)onlineObject);
             if (onlineObject.apo.realizedObject is Overseer) return new RealizedOverseerState((OnlineCreature)onlineObject);
             if (onlineObject.apo.realizedObject is Creature) return new RealizedCreatureState((OnlineCreature)onlineObject);
+            if (onlineObject.apo.realizedObject is Fly) return new RealizedFlyState((OnlineCreature)onlineObject);
```
Comment (henpemaz) [inline Online/State/AbstractCreatureState.cs:25]: Fly is creature so this never runs (the previous line returns)

File: Game/RainMeadow.GameHooks.cs (hunk 1 above)
Comment (Chrometurtle) [reply Game/RainMeadow.GameHooks.cs]: Found a better function to hook into

File: Online/State/AbstractCreatureState.cs (hunk 2 above)
Comment (Chrometurtle) [reply Online/State/AbstractCreatureState.cs:24]: fixed

**Pull request merged**